src/Workspaces/Core/Portable/Utilities/ForegroundThreadDataKind.cs

```diff
@@ -41,6 +41,12 @@ internal static ForegroundThreadDataKind CreateDefault()
 
                     return ForegroundThreadDataKind.JoinableTask;
 
+                case null:
+                    // If we do not have a SynchronizationContext to examine, assume that the current thread is 
```

Comment:
This would mean we assume all background threads are UI threads, which really isn't what we want. If anything, this has to be an opt-in behavior done during SetSite.